service/policy/attributes/attributes.go

```diff
@@ -99,11 +108,30 @@ func (s *AttributesService) UpdateAttribute(ctx context.Context,
 ) (*attributes.UpdateAttributeResponse, error) {
 	rsp := &attributes.UpdateAttributeResponse{}
 
-	a, err := s.dbClient.UpdateAttribute(ctx, req.GetId(), req)
+	attributeID := req.GetId()
+	auditParams := audit.PolicyEventParams{
+		ActionType: audit.ActionTypeUpdate,
+		ObjectType: audit.ObjectTypeAttributeDefinition,
+		ObjectID:   attributeID,
+	}
+
+	original, err := s.dbClient.GetAttribute(ctx, attributeID)
 	if err != nil {
+		s.logger.Audit.PolicyCRUDFailure(ctx, auditParams)
+		return nil, db.StatusifyError(err, db.ErrTextGetRetrievalFailed, slog.String("id", attributeID))
+	}
+
+	updated, err := s.dbClient.UpdateAttribute(ctx, req.GetId(), req)
+	if err != nil {
+		s.logger.Audit.PolicyCRUDFailure(ctx, auditParams)
 		return nil, db.StatusifyError(err, db.ErrTextUpdateFailed, slog.String("id", req.GetId()), slog.String("attribute", req.String()))
 	}
-	rsp.Attribute = a
+
+	auditParams.Original = original
+	auditParams.Updated = updated
```

Comment:
Have you validated you have the expected `updated` policy objects available for the audit diff? I believe most of the update SQL statements only return the ID from the updated row so your expected diff may require an additional roundtrip. If so, that should be fine performance-wise as reads are expected to be thousands to millions more common than writes to policy objects.

Reply:
I'll dig down and double check these here

Reply:
@jakedoublev made those changes to fetch the updated version from the DB again, I noticed that the `deactivate` methods seem to actually return the entire object instead of just the ID so we don't need to fetch again for those methods

https://github.com/opentdf/platform/blob/da4d78873cf4a7baa125647930dbf7963626fd68/service/policy/db/namespaces.go#L306